test/Microsoft.VisualStudio.Validation.Tests/Microsoft.VisualStudio.Validation.Tests.csproj

```diff
@@ -8,20 +8,11 @@
     <PackageReference Include="xunit.runner.visualstudio" />
     <PackageReference Include="xunit" />
   </ItemGroup>
+
   <ItemGroup>
-    <ProjectReference Include="..\..\src\Microsoft.VisualStudio.Validation\Microsoft.VisualStudio.Validation.csproj" />
-  </ItemGroup>
-  <ItemGroup>
-    <Compile Update="TestStrings.Designer.cs">
-      <DesignTime>True</DesignTime>
-      <AutoGen>True</AutoGen>
-      <DependentUpon>TestStrings.resx</DependentUpon>
-    </Compile>
+    <EmbeddedResource Update="TestStrings.resx" EmitFormatMethods="true" AsConstants="true" />
```

Comment:
❔ Was the `AsConstants="true"` desired, or used as a workaround for something?

Reply:
It is desired for this project, because I actually use the resource names themselves in some of my testing.
The reason should interest you: I found that string table lookups were showing up in perf traces when I was calling `Requires.Argument(condition, Strings.SomeString)` because `SomeString` would be retrieved even in success conditions. So I have overloads that allow `Requires.Argument(condition, Strings.ResourceManager, nameof(String.SomeString))` to avoid the runtime cost of success cases. But using `nameof` is laborious, so having `AsConstants` makes it easy.

Reply:
I suppose that could work as well, provided C# always caches the delegate it creates from that lambda.